CMakeLists.txt

```diff
@@ -6,25 +6,40 @@
 
 cmake_minimum_required(VERSION 2.8.5)
 
-set(MYPROJECT "firmware")
-set(ELF ${MYPROJECT}.elf)
-project(${MYPROJECT} C)
+if(BUILD_TYPE STREQUAL "firmware")
+elseif(BUILD_TYPE STREQUAL "bootloader")
+elseif(BUILD_TYPE STREQUAL "test")
+else()
+    message(FATAL_ERROR "Invalid build type: ${BUILD_TYPE}. Build with:
+    cmake .. -DBUILD_TYPE=type   # 'type' can be 'test', 'firmware', or 'bootloader'
+    ")
+endif()
 
+    
+set(MYPROJECT ${BUILD_TYPE})
+project(${MYPROJECT} C)
 
 #-----------------------------------------------------------------------------
 # Options for building
 
-option(BUILD_TESTS "Build unit tests instead of ARM binary." ON)
+option(USE_UECC_LIB "Use micro ECC instead bitcoin's secp256k1 library." ON)
```

Comment:
Did you decide to go with micro ECC library instead of the secp256k1?
Or otherwise shouldn't the default be OFF?